pkg/inventory/inventory-client.go

```diff
@@ -24,455 +24,420 @@ import (
-		status = getObjStatus(pruneIDs, unionObjs)
-	}
-	klog.V(4).Infof("num objects to prune: %d", len(pruneIDs))
-	klog.V(4).Infof("num merged objects to store in inventory: %d", len(unionObjs))
-	wrappedInv, err := cic.InventoryFactoryFunc(clusterInv)
-	if err != nil {
-		return pruneIDs, err
-	}
-	if err = wrappedInv.Store(unionObjs, status); err != nil {
-		return pruneIDs, err
-	}
+// SingleObjectInfo implements the Info interface and also adds a Name field.
+// This assumes the underlying inventory is a singleton object with a single
+// name. Used by UnstructuredInventory, which assumes a single object.
+type SingleObjectInfo struct {
+	SimpleInfo
+	// name of the singleton inventory object
+	name string
+}
 
-	// Update not required when all objects in inventory are the same and
-	// status does not need to be updated. If status is stored, always update the
-	// inventory to store the latest status.
-	if objs.Equal(clusterObjs) && cic.statusPolicy == StatusPolicyNone {
-		return pruneIDs, nil
-	}
+// GetName returns the name of the inventory object
+func (i *SingleObjectInfo) GetName() string {
+	return i.name
+}
 
-	if dryRun.ClientOrServerDryRun() {
-		klog.V(4).Infof("dry-run create inventory object: not created")
-		return pruneIDs, nil
+// NewSingleObjectInfo constructs a new SingleObjectInfo
+func NewSingleObjectInfo(id ID, nn types.NamespacedName) *SingleObjectInfo {
+	return &SingleObjectInfo{
+		SimpleInfo: SimpleInfo{
+			id:        id,
+			namespace: nn.Namespace,
+		},
+		name: nn.Name,
 	}
-	err = wrappedInv.Apply(ctx, cic.dc, cic.mapper, cic.statusPolicy)
-	return pruneIDs, err
 }
 
-// Replace stores the passed objects in the cluster inventory object, or
-// an error if one occurred.
-func (cic *ClusterClient) Replace(ctx context.Context, localInv Info, objs object.ObjMetadataSet, status []actuation.ObjectStatus,
-	dryRun common.DryRunStrategy) error {
-	// Skip entire function for dry-run.
-	if dryRun.ClientOrServerDryRun() {
-		klog.V(4).Infoln("dry-run replace inventory object: not applied")
-		return nil
-	}
-	clusterInv, err := cic.getClusterInventoryInfo(ctx, localInv)
-	if err != nil {
-		return fmt.Errorf("failed to read inventory from cluster: %w", err)
-	}
+type Factory interface {
+	// NewInventory returns an empty initialized inventory object.
+	// This is used in the case that there is no existing object on the cluster.
+	NewInventory(Info) (Inventory, error)
+}
 
-	clusterObjs, err := cic.GetClusterObjs(ctx, localInv)
-	if err != nil {
-		return fmt.Errorf("failed to read inventory objects from cluster: %w", err)
-	}
+// Inventory is an internal representation of the inventory object that is used
+// to track which objects have been applied by the applier. The interface exists
+// so different underlying resource types can be used to implement the inventory.
+type Inventory interface {
+	Info() Info
+	// ObjectRefs returns the list of object references tracked in the inventory
+	ObjectRefs() object.ObjMetadataSet
```

Comment:
Do we want to use "Get- " prefix on these methods too, so the field names don't need to abbreviate "Obj"?

Reply:
Done